src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerReference.cs

```diff
@@ -31,12 +31,14 @@ public virtual string Display
         }
 
         /// <summary>
-        /// Returns true if this reference is an unresolved reference.
+        /// A unique identifier for this analyzer reference.
         /// </summary>
-        public virtual bool IsUnresolved
-        {
-            get { return false; }
-        }
+        /// <remarks>
+        /// Note that this and <see cref="FullPath"/> serve different purposes. An analyzer reference may not
+        /// have a path, but it always has an ID. Further, two analyzer references with different paths may
+        /// represent two copies of the same analyzer, in which case the IDs should also be the same.
```

Comment:
Is this comment true now? Same assembly at different paths will return different Ids for corresponding AnalyzerFileReference instances correct? Is there no way on CoreCLR to get to the assembly full name (or some such unique ID per assembly) without actually loading the assembly?